daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/Reader.scala

```diff
@@ -130,19 +106,13 @@ object Reader extends Reader[(PackageId, DamlLf.ArchivePayload)] {
     }
   }
 
-  protected[this] override def readArchivePayloadOfVersion(
-      hash: PackageId,
-      lf: DamlLf.ArchivePayload,
-      version: LanguageVersion,
-  ): (PackageId, DamlLf.ArchivePayload) = (hash, lf)
-
-  // Archive Reader that just checks package hash.
-  val HashChecker = new Reader[Unit] {
-    override protected[this] def readArchivePayloadOfVersion(
-        hash: PackageId,
-        lf: DamlLf.ArchivePayload,
-        version: LanguageVersion,
-    ): Unit = ()
-  }
-
+  def apply(recursionLimit: Int = PROTOBUF_RECURSION_LIMIT): Reader = new Reader(recursionLimit)
```

Comment:
at that point maybe just make it a case class? it’s immutable anyway afaict

Reply:
dropped.